src/rust/en.reaperscans/src/parser.rs

```diff
@@ -1,395 +1,246 @@
-
-		mangas.push(Manga {
-			id,
-			cover,
-			title,
-			url,
-			viewer: MangaViewer::Scroll,
-			..Default::default()
-		});
+pub fn parse_manga_list(
+	base_url: String,
+	filters: Vec<Filter>,
+	page: i32,
+) -> Result<MangaPageResult> {
+	let mut search_query = String::new();
+	let mut genres = String::new();
+	let mut status = String::new();
+	let mut order_by = String::new();
+
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				search_query = filter.value.as_string()?.read();
+			}
+			FilterType::Genre => {
+				if let Ok(filter_id) = filter.object.get("id").as_string() {
+					match filter.value.as_int().unwrap_or(-1) {
+						1 => {
+							genres.push_str(filter_id.read().as_str());
+							genres.push(',');
+						}
+						_ => continue,
+					}
+				}
+			}
+			FilterType::Select => match filter.name.as_str() {
+				"Status" => match filter.value.as_int().unwrap_or(-1) {
+					0 => status.push_str("All"),
+					1 => status.push_str("Ongoing"),
+					2 => status.push_str("Hiatus"),
+					3 => status.push_str("Dropped"),
+					4 => status.push_str("Completed"),
+					_ => status.push_str("All"),
+				},
+				"Order By" => match filter.value.as_int().unwrap_or(-1) {
+					0 => order_by.push_str("updated_at"),
+					1 => order_by.push_str("total_views"),
+					2 => order_by.push_str("title"),
+					3 => order_by.push_str("created_at"),
+					_ => order_by.push_str("updated_at"),
+				},
+				_ => continue,
+			},
+			_ => continue,
+		}
 	}
 
-	let pagination = html.select("main div[wire:id] div nav span").text().read();
-	let has_more = pagination.contains("Next");
-
-	Ok(MangaPageResult {
-		manga: mangas,
-		has_more,
-	})
-}
-
-pub fn parse_search(base_url: String, query: String) -> Result<MangaPageResult> {
-	let html: Node = create_search_request_object(String::from(&base_url), query)
-		.expect("Reaperscans: Search POST request was unsuccessful");
-
-	let mut mangas: Vec<Manga> = Vec::new();
-	for i in html.select("ul li").array() {
-		let item = i
-			.as_node()
-			.expect("Reaperscans: Failed to parse a manga node. Title could not be displayed");
-		// Search displays both Comics and Novels in that order.
-		if &item.text().read() == "Novels" {
-			break;
-		}
-		let id = item
-			.select("a")
-			.attr("href")
-			.read()
-			.replace(&base_url, "")
-			.replace("/comics/", "");
-		if id.is_empty() {
-			continue;
-		}
-		let cover = item.select("a img").attr("src").read();
-		let title = item.select("a img").attr("alt").read();
-		mangas.push(Manga {
-			id,
-			cover,
-			title,
-			viewer: MangaViewer::Scroll,
-			..Default::default()
-		});
+	if !genres.is_empty() {
+		genres.pop();
 	}
 
-	Ok(MangaPageResult {
-		manga: mangas,
-		has_more: false,
-	})
+	let url = format!("{}/query?page={}&perPage=20&series_type=Comic&query_string={}&order=desc&orderBy={}&adult=true&status={}&tags_ids=[{}]", BASE_API_URL, page, search_query, order_by, status, genres);
```

Comment:
for future reference, it's better to use `QueryParameters` from aidoku-rs helpers, but you can leave this